src/unity/python/turicreate/data_structures/sframe.py

```diff
@@ -3144,8 +3144,18 @@ def add_column(self, data, column_name="", inplace=False):
         [3 rows x 3 columns]
         """
         # Check type for pandas dataframe or SArray?
+        
+        from collections import Iterable
```

Comment:
For these, we should have as many imports at the module level instead of inside a function; here, it can slow it down quite a bit.  Let's put `from collections import Iterable as _Iterable` at the top and then use _Iterable below.